src/nvim/os/fs.c

```diff
@@ -302,7 +302,9 @@ static bool is_executable(const char *name, char **abspath)
 static bool is_executable_ext(const char *name, char **abspath)
   FUNC_ATTR_NONNULL_ARG(1)
 {
-  const bool is_unix_shell = strstr((char *)path_tail(p_sh), "sh") != NULL;
+  const bool is_unix_shell = strstr((char *)path_tail(p_sh), "powershell") == NULL
+                             && strstr((char *)path_tail(p_sh), "pwsh") == NULL
+                             && strstr((char *)path_tail(p_sh), "sh") != NULL;
```

Comment:
No need to cast into `char *` since `path_tail` already returns a char pointer.  In Vim's source, they needed to cast because `gettail`, Vim's equivalent of `path_tail`, returns a `char_u *`<sup>[1]</sup>.

```suggestion
  const bool is_unix_shell = strstr(path_tail(p_sh), "powershell") == NULL
                             && strstr(path_tail(p_sh), "pwsh") == NULL
                             && strstr(path_tail(p_sh), "sh") != NULL;
```

[1]: https://github.com/vim/vim/blob/4ae0057308c59a0dee2b452736366e49a5a98b3a/src/filepath.c#L2886-L2887